department-of-veterans-affairs/caseflow
# Jruuuu/appeals 8207

Resolves [APPEALS-8207](https://vajira.max.gov/browse/APPEALS-8207)

### Description
I made two major changes

**1.  I updated the dependencies_report_service.rb**
- updated and added ALL_DEPENDENCIES_MAP and added keys so when the end user runs a cache command for example: `Rails.cache.write(:degraded_service_banner_bgs, :display)` it updates the dependencies-check api
- Updated the dependencies_report method to find and create a new array with the map(s) that only have the `:display` value
- Then maps though the new array and grabs the keys to displays in the controller in json format

**2. I updated the PerformanceDegradationBanner.jsx** 
- updated states to show 3 different conditions to show banner.
- added an array state for the degradedServices 
- updated the render method to return when a specific condition is true to display
- also added condition statement when degradedServices is greater than 2 to map through and add commas 
- removed tech debt that was no longer being used 

### Acceptance Criteria
- [ ] Systems team member is able to run the rails command for a specific partnered system being down
Verify banner shows correct message for that system

- [ ] System team member is able to run the rails command to enable banner for multiple parentered systems being down
Verify banner shows correct messages for each system

- [ ] System team member is able to run the rails command to disable/enable banner for partnered system without impacting others

### Testing Plan
1. Assuming that APPEALS-8203 has been merged
2. Have front and backend is running
3. open caseflow app, make sure there is no banner showing
4. log into rails console and run `Rails.cache.write(:degraded_service_banner_bgs, :display)`
5. refresh casflow app
6. banner with BGS is displayed
7. repeat for the remaining systems
- `Rails.cache.write(:degraded_service_banner_vbms, :display)`
- `Rails.cache.write(:degraded_service_banner_vva, :display)`
- `Rails.cache.write(:degraded_service_banner_vacols, :display)`
- `Rails.cache.write(:degraded_service_banner_gov_delivery, :display)`
- `Rails.cache.write(:degraded_service_banner_va_dot_gov, :display)`
8. to hide a system in rails console run `Rails.cache.write(:degraded_service_banner_bgs, :hide)`
9. Refresh caseflow app, BGS should no longer be in the banner
10. repeat for remaining systems
- `Rails.cache.write(:degraded_service_banner_vbms, :hide)`
- `Rails.cache.write(:degraded_service_banner_vva, :hide)`
- `Rails.cache.write(:degraded_service_banner_vacols, :hide)`
- `Rails.cache.write(:degraded_service_banner_gov_delivery, :hide)`
- `Rails.cache.write(:degraded_service_banner_va_dot_gov, :hide)`
11. If there is no longer any systems down, banner should no longer appear.



### User Facing Changes

 BEFORE|AFTER
 ---|---
![BEFORE: OLD BANNER](https://user-images.githubusercontent.com/71367882/186214000-489d7cc0-7d73-44d7-a5ba-b0dd2124273b.png)|![ZERO SYSTEMS DOWN](https://user-images.githubusercontent.com/71367882/186214257-e9418c00-8201-47e9-a27c-9ded1555ab10.png)![ONE SYSTEM DOWN](https://user-images.githubusercontent.com/71367882/186214294-1bd62cc1-ff94-42ab-be5a-671cafaf5df5.png)![TWO SYSTEMS DOWN](https://user-images.githubusercontent.com/71367882/186214334-b0acbdff-ceb0-4f58-8166-06b05b209b0c.png)![THREE SYSTEMS DOWN](https://user-images.githubusercontent.com/71367882/186214350-0f3edb87-04fe-4bae-95a5-64b25c756cbc.png)![3 or MORE SYSTEMS DOWN](https://user-images.githubusercontent.com/71367882/186214375-921f837a-da5f-46c8-b647-c1da654bf2dc.png)
![BEFORE: HIDE OLD BANNER](https://user-images.githubusercontent.com/71367882/186214040-a54964de-1e92-4d2d-8d17-71f3936e37f4.png)|![HIDE TWO SYSTEMS](https://user-images.githubusercontent.com/71367882/186218148-f429f214-fd8f-4cba-af3e-51af4c01ec5e.png)![HIDE 4 SYSTEMS](https://user-images.githubusercontent.com/71367882/186218212-3239689d-ab9a-4f2e-b4f1-3adb50815548.png)![HIDE 5 SYSTEMS ](https://user-images.githubusercontent.com/71367882/186218242-73430a21-cce2-4021-9237-b34aa96be4a7.png)![HIDE ALL SYSTEMS](https://user-images.githubusercontent.com/71367882/186218277-eed752db-76e4-41f7-91f2-71a8fa82facc.png)





### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*

### Database Changes
*Only for Schema Changes*


### Integrations: Adding endpoints for external APIs
* [ ] Integrations impacting functionality are tested in Caseflow UAT
